sdk/lib/rtl/path.c

```diff
@@ -459,12 +464,130 @@ RtlGetLengthWithoutLastFullDosOrNtPathElement(IN ULONG Flags,
 
 NTSTATUS
 NTAPI
-RtlComputePrivatizedDllName_U(IN PUNICODE_STRING DllName,
-                              IN PUNICODE_STRING a2,
-                              IN PUNICODE_STRING a3)
+RtlComputePrivatizedDllName_U(
+    _In_ PUNICODE_STRING DllName,
+    _Out_ PUNICODE_STRING RealName,
+    _Out_ PUNICODE_STRING LocalName)
 {
-    UNIMPLEMENTED;
-    return STATUS_NOT_IMPLEMENTED;
+    static const UNICODE_STRING ExtensionChar = RTL_CONSTANT_STRING(L".");
+
+    USHORT Position;
+    UNICODE_STRING ImagePathName, DllNameOnly, CopyRealName, CopyLocalName;
+    BOOLEAN HasExtension;
+    ULONG RequiredSize;
+    NTSTATUS Status;
+    C_ASSERT(sizeof(UNICODE_NULL) == sizeof(WCHAR));
+
+    CopyRealName = *RealName;
+    CopyLocalName = *LocalName;
+
+
+    /* Get the image path */
+    ImagePathName = RtlGetCurrentPeb()->ProcessParameters->ImagePathName;
+
+    /* Check if it's not normalized */
+    if (!(RtlGetCurrentPeb()->ProcessParameters->Flags & RTL_USER_PROCESS_PARAMETERS_NORMALIZED))
+    {
+        /* Normalize it*/
+        ImagePathName.Buffer = (PWSTR)((ULONG_PTR)ImagePathName.Buffer + (ULONG_PTR)RtlGetCurrentPeb()->ProcessParameters);
+    }
+
+
+    if (!NT_SUCCESS(RtlFindCharInUnicodeString(RTL_FIND_CHAR_IN_UNICODE_STRING_START_AT_END,
+                                               DllName, &RtlpPathDividers, &Position)))
+    {
+        DllNameOnly = *DllName;
+    }
+    else
+    {
+        /* Just keep the dll name, ignore path components */
+        Position += sizeof(WCHAR);
+        DllNameOnly.Buffer = DllName->Buffer + Position / sizeof(WCHAR);
+        DllNameOnly.Length = DllName->Length - Position;
+        DllNameOnly.MaximumLength = DllName->MaximumLength - Position;
+    }
+
+    if (!NT_SUCCESS(RtlFindCharInUnicodeString(RTL_FIND_CHAR_IN_UNICODE_STRING_START_AT_END,
+                                               &DllNameOnly, &ExtensionChar, &Position)))
+    {
+        Position = 0;
+    }
+
+    HasExtension = Position > 1;
+
+    /* First we create the c:\path\processname.exe.Local\something.dll path */
+    RequiredSize = ImagePathName.Length + RtlpDotLocal.Length + DllNameOnly.Length +
+        (HasExtension ? 0 : RtlpDefaultExtension.Length) + sizeof(UNICODE_NULL);
+
+    /* This is not going to work out */
+    if (RequiredSize >= UNICODE_STRING_MAX_BYTES)
```

Comment:
Why fail on equality?

Reply:
I think that originally I wrote this the other way around, and simply failed reversing the condition or so.
I'll fix it.